### ISSUE TITLE
Disable consider-using-join for non-empty separator

### ISSUE THREAD

**Issue by vmeurisse** [pylint-dev/pylint#8701]
### Current problem

In 2.16.0, consider-using-join was extended to include non-empty separators. This makes pylint complains about the following code:

```py
def test(values: List[str]):
    data = ""
    for value in values:
        data += f"prefix{value}suffix"
    return data
```

However I don't think the suggested refactor is an improvement: prefix and suffix are duplicated and the empty list case need to be handled specifically

```py
def test(values: List[str]):
    if not values:
        return ""
    return "prefix" + "suffixprefix".join(values) + "suffix"
```

The current options are:
 - disable the rule completely. This has the drawback of loosing the previous rule for empty separators.
 - use a "pylint: disable" comment. However having to use this on regular basis creates bad habits and reduce the value of linting

### Desired solution

Would it be possible to have an option to disable the rule only for the non-empty separator case?

**Comment by Pierre-Sassoulas** [comment]
I remember having the same sentiment during initial implementation too. I think adding an option make sense.

**Comment by vivekthedev** [comment]
I am interested in solving this one can you guide me through it?

**Comment by vivekthedev** [comment]
Hey @Pierre-Sassoulas  Thanks for providing the code sections. I have created the option but how should I go about modifying the checking. Should I write a new function or detect non-empty separator in the same function and return the function if found. I am new to contributing to large codebases.

**Comment by Pierre-Sassoulas** [comment]
I think the ``_check_consider_using_join`` function can be tweaked a little to detect the non empty separator we want to avoid raising for if the ``suggest-join-with-non-empty-separator`` is true.

**Comment by vivekthedev** [comment]
Hey @Pierre-Sassoulas  Apologies for late response. In this problem do we have to check the code for non-empty separators with astroid or just adding the the if statement like this is enough:
```
 if self.linter.config.suggest_join_with_non_empty_separator:
            return
```

**Comment by theirix** [comment]
Hello. I have tried to implement this feature, please take a look.
